python/cost_models.py

```diff
@@ -0,0 +1,220 @@
+    # relatively hard, whereas it treats large residuals comparably easy.
+    LNQ = 1/len(y_true) * np.sum(np.exp(np.divide(y_pred, y_true)))
+
+    # Percentage
+    perc = np.mean(100 * (np.divide(np.abs(y_true - y_pred), y_true)))
+
+    scores = {'MSE': '%.3f' % mse, 'R2': '%.3f' % R2, 'LNQ': '%.3f' % LNQ, 'Percentage': '%.3f' % perc}
+
+    return scores
+
+
+def log(scores, out):
+    with open('{}/log.txt'.format(out), 'w') as file:
+        for entry in scores:
+            file.write(entry + ': ' + str(scores[entry]) +  '\n')
+
+
+# needed for prediction with one-hot-encoding in case trainings and test data don't have the same set of values in a
+# categorical data column
+def add_dummy_types(train, test, cols):
+    for col in cols:
+        diff1 = np.setdiff1d(train[col].unique(), test[col].unique())
+        diff2 = np.setdiff1d(test[col].unique(), train[col].unique())
+        for d1 in diff1:
+            test['{}_{}'.format(col, d1)] = 0
+        for d2 in diff2:
+            train['{}_{}'.format(col, d2)]= 0
+    return [train, test]
+
+
+def parseargs(opt=None):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('-train', help='Trainingsdata in csv format', action='append', nargs='+')
+    # in case no test data is given, the trainings data will be split into trainings and test data
+    parser.add_argument('--test', help='Testdata in csv format', action='append', nargs='+')
+    parser.add_argument('--m', help='Model type: choose from "linear", "lasso", "ridge", "boost"; Boost is the default"')
+    parser.add_argument('--out', help='Output folder')
+
+    if (opt):
+        return parser.parse_args(opt)
+    else:
+        return parser.parse_args()
+
+
+def import_data(args):
+    test_data = import_train_data(args.test[0])
+    test_data = test_data.dropna()
+
+    train_data = import_train_data(args.train[0])
+    train_data = train_data.dropna()
+
+    # check whether trainings and testdata have the same format
+    if test_data.columns.all() != train_data.columns.all():
+        warnings.warn("Warning: Trainings- and Testdata do not have the same format. Unmatched columns will be ignored!")
+
+    inter = train_data.columns.intersection(test_data.columns)
+    test_data = test_data[inter.tolist()]
+    train_data = train_data[inter.tolist()]
+
+    return [train_data, test_data]
+
+
+def main(args):
+    out = "CostModelOutput"
+    scores = {}
+
+    if args.m:
+        model_types = [args.m]
+    else:
+        # We have achieved the best results with this model
+        model_types = ['boost']
+
+    if args.test:
+        train_data, test_data = import_data(args)
+    else:
+        train_data = import_train_data(args.train[0])
+        train_data = train_data.dropna()
+        train_data, test_data = train_test_split(train_data)
+
+    if args.out:
+        out = args.out
+
+    if not os.path.exists("{}/Models".format(out)):
```

Comment:
Better avoid uppercase letters in folder names.

Reply:
Done